### PR TITLE
sendAnonymousCLIMetrics added

### PR DESCRIPTION
**Pull request by lazywithclass**
Send metrics to the registry as per #138

### CODE REVIEW

**Comment by iarna** [comment]
Finally, FINALLY, we're ready to take this! =D

File: lib/sendAnonymousCLIMetrics.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module.exports = send
+
+var assert = require('assert')
+var url = require('url')
+
+function send (uri, params, cb) {
```
Comment (iarna) [inline lib/sendAnonymousCLIMetrics.js]: This function should compose its URL on its own (for example how `lib/dist-tags/set.js` works).

File: test/send-anon-metrics.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+var test = require('tap').test
+
+var common = require('./lib/common.js')
+var client = common.freshClient()
+var server = require('./lib/server.js')
+
+function nop () {}
+
+var URI = 'https://npm.registry:8043/-/npm/anon-metrics/v1/:metricId'
```
Comment (iarna) [inline test/send-anon-metrics.js]: `:metricId` shouldn't be passed in literally, it's supposed to be replaced with the `metricId` value from the document we're given.

File: lib/sendAnonymousCLIMetrics.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  assert(typeof cb === 'function', 'must pass callback')
+
+  var parsed = url.parse(uri)
+  assert(
+    parsed.protocol === 'http:' || parsed.protocol === 'https:',
+    'must have a URL that starts with http: or https:'
+  )
+
+  params.method = 'PUT'
+  this.request(uri, params, cb)
```
Comment (iarna) [inline lib/sendAnonymousCLIMetrics.js]: Params here can't just be passed through to `request`.  What you're passing in the test isn't valid for `this.request`. Again, see `lib/dist-tags/set.js`, they would need to be JSON encoded and put into a `body` property.

**Comment by lazywithclass** [comment]
hehe ok that's good, I'll have a look at the changes requested. If you're going to be faster, I will just compare the results and see how this should've been done. Thanks nonetheless.

**Comment by iarna** [comment]
Merged as 9c5db49

**Pull request closed without merging**